src/fsharp/service/FSharpParseFileResults.fsi

```diff
@@ -46,6 +46,9 @@ type public FSharpParseFileResults =
     /// </summary>
     member TryRangeOfRefCellDereferenceContainingPos: expressionPos: pos -> range option
 
+    /// Gets the range of an expression being dereferenced. For `!expr`, gives the range of `expr`
+    member TryRangeOfExpressionBeingDereferencedContainingPos: expressionPos: pos -> range option
```

Comment:
A more efficient approach is to unify this with `TryRangeOfRefCellDereferenceContainingPos`. This would be a breaking change, but we're already doing several breaks in FCS 40.